lib/Page.js

```diff
@@ -775,20 +775,21 @@ class Page extends EventEmitter {
    * @param {!Array<string>} values
    */
   async select(selector, ...values) {
-    await this.$eval(selector, (element, values) => {
+    return await this.$eval(selector, (element, values) => {
       if (element.nodeName.toLowerCase() !== 'select')
         throw new Error('Element is not a <select> element.');
-
       const options = Array.from(element.options);
-
+      const filteredValues = values.filter(value => options.some(option => option.value === value));
       if (element.multiple) {
         for (const option of options)
           option.selected = values.includes(option.value);
       } else {
-        element.value = values.shift();
+        filteredValues.splice(1);
```

Comment:
if we want to return values that are selected, i'd prefer generating them from options.

```js
// ... 
return options.filter(option => option.selected).map(option => option.value);
```